documents/invoice/model.go

```diff
@@ -601,6 +601,29 @@ func (i *Invoice) unpackFromUpdatePayload(old *Invoice, payload documents.Update
 	return nil
 }
 
+// unpackFromUpdatePayloadOld unpacks the update payload and prepares a new version.
+func (i *Invoice) unpackFromUpdatePayload(payload documents.UpdatePayload) (*Invoice, error) {
+	var d Data
+	err := copier.Copy(&d, &i.Data)
```

Comment:
deep copy for complex structs?

Reply:
yes, this way we dont have to old invoice data which on memory doesn't share the same data with the new one. One use case for this is validators